# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentChannelParent.cpp
# Commit: 0260b9ef8c6c
# Full Hash: 0260b9ef8c6cf2059f3024e798a2a58680ff9ee7
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-07-31 09:46:06
# Description:
#   Bug 1656039 - Fall back to content initiating the load if parent went away. r=mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D85519
# ==============================================================================

diff -r 164f54ec7aff -r 0260b9ef8c6c netwerk/ipc/DocumentChannelParent.cpp
--- a/netwerk/ipc/DocumentChannelParent.cpp	Fri Jul 31 01:45:44 2020 +0000
+++ b/netwerk/ipc/DocumentChannelParent.cpp	Fri Jul 31 03:59:10 2020 +0000
@@ -40,10 +40,8 @@
   if (loadState->GetChannelInitialized()) {
     promise = DocumentLoadListener::ClaimParentLoad(
         getter_AddRefs(mDocumentLoadListener), loadState->GetLoadIdentifier());
-    if (!promise) {
-      return false;
-    }
-  } else {
+  }
+  if (!promise) {
     bool isDocumentLoad =
         aArgs.elementCreationArgs().type() ==
         DocumentChannelElementCreationArgs::TDocumentCreationArgs;